pkg/controller/baremetalhost/host_state_machine.go

```diff
@@ -88,6 +88,25 @@ func (hsm *hostStateMachine) updateHostStateFrom(initialState metal3v1alpha1.Pro
 			stateChanges.With(stateChangeMetricLabels(initialState, hsm.NextState)).Inc()
 		})
 		hsm.Host.Status.Provisioning.State = hsm.NextState
+		// Here we assume that if we're being asked to change the
+		// state, the return value of ReconcileState (our caller) is
+		// set up to ensure the change in the host is written back to
+		// the API. That means we can safely update any status fields
+		// along with the state.
+		switch hsm.NextState {
+		case metal3v1alpha1.StateRegistering,
```

Comment:
Is there any point doing this before Registering when it's not copied to Ironic until we reach Inspecting?

I don't think it's safe to do this before #388 is merged, because right now this could end up changing the status of a provisioned Host so that the status would no longer match what we actually provisioned with.

Reply:
I can remove registering from the list here.